### PR TITLE
Escape donate link output in plugin_row_meta

### PR DESCRIPTION
**Pull request by obenland**
## Summary

- Wrap `$this->donate_link` in `esc_url()` and the translated `Donate` label in `esc_attr__()` inside `Obenland_Wp_Plugins_V5::plugin_row_meta()` so a tampered `.mo` translation or any future change exposing `donate_link` to external input can't break out of the `href`/`title` attributes.
- The values are server-controlled today, so this is defense-in-depth — surfaced by a security audit pass over the codebase.
- Add four PHPUnit tests covering the happy path, the non-matching plugin pass-through, and two negative cases that inject malicious payloads (`javascript:alert(1)" onclick="alert(2)` via reflection, `"><script>alert(1)</script>` via the `gettext` filter) and assert they don't survive escaping.

## Test plan

- [x] `composer test` — single-site, 20/20 passing
- [x] `composer test-multisite` — 20/20 passing
- [x] `vendor/bin/phpcs --standard=WordPress` on changed files — clean
- [x] Reverted the fix locally and confirmed both XSS tests fail with the unescaped payloads visible in output, then re-applied and confirmed they pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: class-obenland-wp-plugins-v5.php [hunk 1]
```diff
@@ -125,8 +125,8 @@ public function plugin_row_meta( $plugin_meta, $plugin_file ) {
 		if ( $this->plugin_name === $plugin_file ) {
 			$plugin_meta[] = sprintf(
 				'<a href="%1$s" target="_blank" title="%2$s">%2$s</a>',
-				$this->donate_link,
-				__( 'Donate', 'obenland-wp' )
+				esc_url( $this->donate_link ),
+				esc_attr__( 'Donate', 'obenland-wp' )
 			);
```
[the remark this answers is not in the document]
Comment (obenland) [reply class-obenland-wp-plugins-v5.php:132]: Fixed — split into separate placeholders so the `title` attribute uses `esc_attr()` and the anchor text uses `esc_html()`. The translated label is captured once in a local to avoid two `__()` calls.

File: class-obenland-wp-plugins-v5.php [hunk 2]
```diff
@@ -125,8 +125,8 @@ public function plugin_row_meta( $plugin_meta, $plugin_file ) {
 		if ( $this->plugin_name === $plugin_file ) {
 			$plugin_meta[] = sprintf(
 				'<a href="%1$s" target="_blank" title="%2$s">%2$s</a>',
```
[the remark this answers is not in the document]
Comment (obenland) [reply class-obenland-wp-plugins-v5.php]: Fixed — added `rel="noopener noreferrer"` to the anchor and asserted it in the happy-path test so it cannot regress silently.
